From 7971432c7f4b8e98aa808fc8a205b536b647a4b5 Mon Sep 17 00:00:00 2001
From: Zhangbin Tong <zebulun.tong@rock-chips.com>
Date: Mon, 15 Oct 2018 11:11:44 +0800
Subject: [PATCH] arm: dts: rk3128h-box-avb: Add the possible "boot devices"

During uevent processing, some "by-name" symlinks will be created.
The following symlinks will then be created on the device.

	/dev/block/by-name/<partition> -> /dev/block/<type>/<device>/by-name/<partition>

Note that both <type> and <device> are skipped in the newly create symlinks.
It assumes there is no more than one devices with the same <partition>,
which is the assumption of current first stage mount flow.

When 'boot_devices' in DT is absent, it fallbacks to extract
'boot_devices' from fstab settings.

Change-Id: Ie7f721b1ae53245c324f7ef2c707e7776aeacf66
Signed-off-by: Zhangbin Tong <zebulun.tong@rock-chips.com>
---
 arch/arm/boot/dts/rk3128h-box-avb.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/rk3128h-box-avb.dts b/arch/arm/boot/dts/rk3128h-box-avb.dts
index a480ba513417..7cb5f92bb9da 100644
--- a/arch/arm/boot/dts/rk3128h-box-avb.dts
+++ b/arch/arm/boot/dts/rk3128h-box-avb.dts
@@ -36,6 +36,7 @@
 
 &firmware_android {
 	compatible = "android,firmware";
+	boot_devices = "30020000.dwmmc,30030000.nandc";
 	vbmeta {
 		compatible = "android,vbmeta";
 		parts = "vbmeta,boot,system,vendor,dtbo";
@@ -44,7 +45,7 @@
 		compatible = "android,fstab";
 		vendor {
 			compatible = "android,vendor";
-			dev = "/dev/block/platform/30020000.dwmmc/by-name/vendor";
+			dev = "/dev/block/by-name/vendor";
 			type = "ext4";
 			mnt_flags = "ro,barrier=1,inode_readahead_blks=8";
 			fsmgr_flags = "wait,avb";
-- 
2.35.3

